pkg/generate/dockercompose/lookup.go

```diff
@@ -0,0 +1,57 @@
+package dockercompose
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/golang/glog"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	"github.com/openshift/origin/pkg/generate/app"
+)
+
+// TemplateFileSearcher resolves template files into template objects
+type FileSearcher struct {
+}
+
+// Search attemps to read template files and transform it into template objects
+func (r *FileSearcher) Search(precise bool, terms ...string) (app.ComponentMatches, []error) {
+	matches := app.ComponentMatches{}
+	var errs []error
+	for _, term := range terms {
+		if term == "__dockercomposefile_fail" {
+			errs = append(errs, fmt.Errorf("unable to find the specified template file: %s", term))
+			continue
+		}
+		if !IsPossibleDockerCompose(term) {
+			continue
+		}
+		if _, err := os.Stat(term); err != nil {
+			continue
+		}
+		template, err := Generate(term)
```

Comment:
doing this inside search seems awkward/expensive, over returning potential matching files and then only invoking Generate on the best/only match, if there is one (if there are multiple matches in most/all cases these days we return a warning to the user rather than taking any action)